drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/document_exports/report_chart_pdf_export.rb

```diff
@@ -0,0 +1,74 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisDataQualityTool::DocumentExports
+  class ReportChartPdfExport < ::GrdaWarehouse::DocumentExport
+    include ApplicationHelper
+    def authorized?
+      user.can_view_any_reports? && report_class.viewable_by(user)
+    end
+
+    protected def report
+      @report ||= report_class.find(params['id'])
+    end
+
+    protected def view_assigns
+      {
+        report: report,
+        filter: filter,
+        title: Translation.translate('HMIS Data Quality Tool'),
+        pdf: true,
+        # Ensure we have per-page JS
+        per_page_js: [
+          'hmis_dq_tool_completeness',
+          'hmis_dq_tool_time_to_enter',
+          'hmis_dq_tool_time_in_enrollment',
+        ],
+      }
+    end
+
+    def perform
```

Comment:
the perform method seems to be repeated  or similar across 7 different DocumentExports. Perhaps this should be dried up at some point.